astropy/table/table.py

```diff
@@ -856,9 +856,17 @@ def pprint(self, max_lines=None, max_width=None, show_name=True,
             else:
                 print(line)
 
+    def _make_index_row_display_table(self, index_row_name='idx'):
+        if index_row_name not in self.columns:
```

Comment:
Hmm, I see your point that you want to be able to get this sometimes, but that solution worries me for cases where "idx" is already in a table: I suspect the user (which is *me* in some cases I'm working with right now...) would get confused about which "idx" is the one that's actually in the table and which is auto-generated.  So I prefer that it gets hidden when "idx" is present.

I just pushed up what I think is probably a better solution anyway, though (see below).